src/lib/routers/__tests__/history.test.ts

```diff
@@ -0,0 +1,96 @@
+import historyRouter from '../history';
+
+const wait = (ms = 0) => new Promise(res => setTimeout(res, ms));
+
+describe('life cycle', () => {
+  beforeEach(() => {
+    window.history.pushState(null, '-- divider --', 'http://localhost/');
+    jest.restoreAllMocks();
+  });
+
+  it('does not write the same url twice', async () => {
+    const pushState = jest.spyOn(window.history, 'pushState');
+    const router = historyRouter({
+      writeDelay: 0,
+    });
+
+    router.write({ some: 'state' });
+    await wait(0);
+
+    router.write({ some: 'state' });
+    await wait(0);
+
+    router.write({ some: 'state' });
+    await wait(0);
+
+    expect(pushState).toHaveBeenCalledTimes(1);
```

Comment:
What is the expected history state (or what was `pushState` called with)? We can specify it here.